### PR TITLE
Seafaring tutorial improvements

### PR DESCRIPTION
**Pull request by pospec4444**
- Start the Ship Yard objective while the port is still building
- Added Ferry Yard objective (to make a break in ferrying tutorial text)
- Scroll to gold resource indicator

Issue #3675

### CODE REVIEW

File: data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua [hunk 1]
```diff
@@ -73,10 +73,11 @@ tell_about_port_building = {
 
 tell_about_shipyard = {
    position = "topright",
+   field = shipyard_tip,
    title = _"Constructing ships",
    body = (
       h1(_"Let’s build ships") ..
-      p(_[[Great. Your port has just been finished. Now we need some ships.]]) ..
+      p(_[[It seems like port construction is in progress. Meanwhile, let's build some ships.]]) ..
```
Comment (gunchleoc) [inline data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua]:
```suggestion
      p(_[[Great. While your port is being constructed, let’s build some ships.]]) ..
```

File: data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua [hunk 2]
```diff
@@ -187,28 +188,37 @@ ferry_1 = {
...
-      li_image("images/wui/fieldaction/menu_build_water.png", _[[Unlike ships which travel to any port where they’re needed, ferries have fixed transport routes called waterways. A waterway is built just like a road: You click on a flag near the shore, choose ‘Build a waterway’, and select the path for the waterway just as you would do for a road.]]) ..
-      p(_[[Waterways can be built on the open sea as well as close to the shore where the water is too shallow for big ships. The only rule is that the two triangles directly adjacent to each segment of the waterway have to be water, and the entire path needs to be within your territory.]])
+      h1(_"Rowboats & Ferry Yards") ..
+      li_object("atlanteans_ferry_yard", _[[This is where ferries come in. Ferries are small rowboats that can carry only one ware at a time, and no workers (other than the one who rows it and who never leaves his boat). Ferries are built in ferry yards, one rowboat costs two logs and one piece of cloth and is quickly constructed. A ferry yard is a medium-sized building that needs to be built close to the shore.]]) ..
+      li(_[[Let's build a ferry yard close to the shore before we will continue.]])
```
Comment (gunchleoc) [inline data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua]:
```suggestion
      li(_[[Let’s build a ferry yard close to the shore before we will continue.]])
```

File: data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua [hunk 3]
```diff
@@ -187,28 +188,37 @@ ferry_1 = {
...
-      li_image("images/wui/fieldaction/menu_build_water.png", _[[Unlike ships which travel to any port where they’re needed, ferries have fixed transport routes called waterways. A waterway is built just like a road: You click on a flag near the shore, choose ‘Build a waterway’, and select the path for the waterway just as you would do for a road.]]) ..
-      p(_[[Waterways can be built on the open sea as well as close to the shore where the water is too shallow for big ships. The only rule is that the two triangles directly adjacent to each segment of the waterway have to be water, and the entire path needs to be within your territory.]])
+      h1(_"Rowboats & Ferry Yards") ..
+      li_object("atlanteans_ferry_yard", _[[This is where ferries come in. Ferries are small rowboats that can carry only one ware at a time, and no workers (other than the one who rows it and who never leaves his boat). Ferries are built in ferry yards, one rowboat costs two logs and one piece of cloth and is quickly constructed. A ferry yard is a medium-sized building that needs to be built close to the shore.]]) ..
```
Comment (gunchleoc) [inline data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua]: The text is a bit long. You can split it up into multiple p() within the li_object.

**Comment by gunchleoc** [comment]
Code is OK, just some proofreading suggestions.

**Comment by pospec4444** [comment]
There is one more thing about this tutorial.

Motivation to build a colony on the island is ore. But there is no follow-up on this goal. Moreover, when player is not slacking and follows instructions, colony will be finished at almost the same time as second ship is built.

At that time, ferrying tutorial starts and shifts focus on southern part of the map. Player will easily forget about stopping shipyard construction. Nothing motivates him to build iron mine on the island.

Shouldn't there be another objective covering iron mining on the island? This would start after colony is built and before ferrying part.

**Review by gunchleoc — APPROVED** [review]

**Comment by gunchleoc** [comment]
I like your ideas - please go ahead and be as creative as you want :)

BTW we can now count how many ships a player has https://www.widelands.org/documentation/autogen_wl_game/#player you could use that to tell the player to stop the shipyard when the 2 ships are finished, and to trigger the expedition objective after the first ship is done.

**Comment by pospec4444** [comment]
OK. Shall I update original issue or is it fine to track my ideas here?

My TODO:

1. Add new objective to build an iron mine on the island after third port is built
2. ~~Start the expedition objective when first ship is done
(after some "lead time" to allow few trips between north and south?)~~ (already done)
3. Remind player to stop shipyard production when second ship is built
4. If possible, remind player to stop ferry yard production when there is enough ferries built
(probably use [get_workers()](https://www.widelands.org/documentation/autogen_wl_bases/?highlight=get_workers#wl.bases.PlayerBase.get_workers) for that)

Use [get_ships()](https://www.widelands.org/documentation/autogen_wl_game/#wl.game.Player.get_ships) for 2. and 3.

**Comment by gunchleoc** [comment]
Sounds good.

All the ferry stuff could also be done while waiting for the port and ships to be built, so the player won't have to wait that long.

I'll merge this branch now - you can create a follow-up branch :)

**Pull request merged**